File: Training/Classic Puzzle - Easy/The_Descent.java
/**
 * @file The_Descent.java
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-12-25
 * 
 * @copyright Copyright (c) 2021
 */
 
import java.util.*;

class Player {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        // game loop
        while (true) {
            int index = -1, maxHeight = -1;
            for (int idx = 0; idx < 8; ++idx) {
                int mountainH = in.nextInt(); // represents the height of one mountain.
                if (mountainH > maxHeight){
                    maxHeight = mountainH;
                    index = idx;
                }
            }
            System.out.println(index); // The index of the mountain to fire on.
        }
    }
}

File: Training/Classic Puzzle - Easy/The_Descent.c
/**
 * @file The_Descent.c
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-12-25
 * 
 * @copyright Copyright (c) 2021
 */
 
#include <stdio.h>

int main(){
    // game loop
    while (1) {
        int index = -1, maxHeight = -1;
        for (int idx = 0; idx < 8; ++idx) {
            // represents the height of one mountain.
            int mountain_h;
            scanf("%d", &mountain_h);
            if (mountain_h > maxHeight) {
                maxHeight = mountain_h;
                index = idx;
            }
        }

        printf("%d\n", index); // The index of the mountain to fire on.
    }
    return 0;
}